src/Table/index.jsx

```diff
@@ -1,16 +1,88 @@
 import React from 'react';
 import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import isRequiredIf from 'react-proptype-conditional-require';
 import PropTypes from 'prop-types';
 
 import styles from './Table.scss';
+import Button from '../Button';
 
 class Table extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sortedColumn: props.tableSortable ? this.props.defaultSortedColumn : '',
+      sortDirection: props.tableSortable ? this.props.defaultSortDirection : '',
+    };
+
+    this.onSortClick = this.onSortClick.bind(this);
+  }
+
+  onSortClick(columnKey) {
+    let newDirection = 'desc';
+
+    if (this.state.sortedColumn === columnKey) {
+      newDirection = (this.state.sortDirection === 'desc' ? 'asc' : 'desc');
+    }
+
+    this.setState({
+      sortedColumn: columnKey,
+      sortDirection: newDirection,
+    });
+
+    const currentlySortedColumn = this.props.columns.find(
+      column => (columnKey === column.key));
+    currentlySortedColumn.onSort(newDirection);
+  }
+
   getCaption() {
     return this.props.caption && (
       <caption>{this.props.caption}</caption>
     );
   }
 
+  getSortButtonScreenReaderText(columnKey) {
+    let text = '';
+
+    if (this.state.sortedColumn === columnKey) {
+      text = this.props.sortButtonsScreenReaderText[this.state.sortDirection];
+    } else {
+      text = this.props.sortButtonsScreenReaderText[''];
```

Comment:
This is clever code! But it's not especially readable. Can you use a more explicit keyname for the default state in `sortButtonsScreenReaderText`, like `defaultText` or something?